main.py

```diff
@@ -0,0 +1,46 @@
+"""
+Entry point for LTE Communication.
+"""
+import time
+
+from modules.common.mavlink.modules import flight_controller
+
+DELAY_TIME = 1.0  # seconds
+MISSION_PLANNER_ADDRESS = "tcp:127.0.0.1:14550"
+
+# Main Function
+def main() -> int:
+    """
+    Main function for LTE Communication code. Assumes that a mission has already been uploaded to 
+    the drone at MISSION_PLANNER_ADDRESS.
+    """
+    result, controller = flight_controller.FlightController.create(MISSION_PLANNER_ADDRESS)
+    if not result:
+        print("Failed to create flight controller.")
+        return -1
+
+    print("Entering polling loop for drone end of mission.")
+
+    while True:
+        result, is_mission_end = controller.is_drone_destination_final_waypoint()
```

Comment:
`is_drone_destination_final_waypoint()` does not indicate the drone is actually at the final waypoint; it indicates the drone's intended destination is the final waypoint. The drone could be far away in a different mode, or heading towards the final waypoint, or hovering over the final waypoint.

Is this the intended logic?

Reply:
To quote Mihir: 
"I have the method in the FlightController class in the common repository
And an example of the while loop in test_mission_ended
So we want to basically implement the same while loop in the image-transmission library
The reason we want to do this is because we don’t want to start sending images when the drone turns on, that would be spend too much data
Instead we want to start to send images when the drone is near the end of the mission
So this while loop will let the program know when the drone is close enough"

Reply:
I guess the drone is "close enough" when it's current destination is the final waypoint.

Reply:
Please confirm with Mihir that this is the case, as the final task 2 CONOPS waypoint is away from the landing zones, and the drone must wait to approach the landing zone until the correct time window.

Reply:
Ok I'll confirm and let you know in this comment thread